pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -254,11 +256,26 @@ func validateServiceInstanceUpdate(instance *sc.ServiceInstance) field.ErrorList
 }
 
 // internalValidateServiceInstanceUpdateAllowed ensures there is not a
-// pending update on-going with the spec of the instance before allowing an update
-// to the spec to go through.
+// pending update on-going with the spec of the instance before allowing an
+// update to the spec to go through unless its the same user who made the
+// original update.
 func internalValidateServiceInstanceUpdateAllowed(new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
 	errors := field.ErrorList{}
-	if old.Generation != new.Generation && old.Status.CurrentOperation != "" {
+
+	oldUID := ""
+	newUID := ""
+
+	if old.Spec.UserInfo != nil {
+		oldUID = old.Spec.UserInfo.UID
+	}
+
+	if new.Spec.UserInfo != nil {
+		newUID = new.Spec.UserInfo.UID
+	}
+
+	disableLocking := utilfeature.DefaultFeatureGate.Enabled(scfeatures.DisableInstanceLocking)
```

Comment:
if `disableInstanceLocking` is true, we can just return true at the beginning of this func